config/config.go

```diff
@@ -1301,3 +1307,28 @@ func (c *RemoteWriteConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	}
 	return nil
 }
+
+// RemoteReadConfig is the configuration for reading from remote storage.
+type RemoteReadConfig struct {
+	URL           *URL           `yaml:"url,omitempty"`
+	RemoteTimeout model.Duration `yaml:"remote_timeout,omitempty"`
+	BasicAuth     *BasicAuth     `yaml:"basic_auth,omitempty"`
+	TLSConfig     TLSConfig      `yaml:"tls_config,omitempty"`
+	ProxyURL      URL            `yaml:"proxy_url,omitempty"`
```

Comment:
There's actually a `HTTPClientConfig` in `config` already, so we probably shouldn't create yet another similar HTTP client config struct for embedding here. That one supports more HTTP auth methods (bearer token, ...), but lacks the timeout and url.

@brian-brazil Was there a specific reason why we only do basic auth for remote write, or would it be possible to reuse the `HTTPClientConfig` here and for the write path as well?